frontend/src/metabase/visualizations/echarts/cartesian/option/trend-line.ts

```diff
@@ -0,0 +1,195 @@
+    })),
+  );
+}
+
+/**
+ * Applies a square root function to all values in the datasets for all trend lines,
+ * for use with the power y-axis scale.
+ *
+ * @param {TrendDataset[]} trendDatasets - Datasets for the trend lines for all series in the chart.
+ * @param {ComputedVisualizationSettings} settings - Locally computed visualization settings for the chart
+ * @returns {TrendDataset[]} Square-rooted datasets if the `graph.y_axis.scale` setting is `pow`,
+ * otherwise the unchanged input datasets
+ */
+function squareRootScaleDatasets(
+  trendDatasets: TrendDataset[],
+  settings: ComputedVisualizationSettings,
+): TrendDataset[] {
+  if (settings["graph.y_axis.scale"] !== "pow") {
+    return trendDatasets;
+  }
+
+  return trendDatasets.map(trendDataset =>
+    replaceValues(trendDataset, (dataKey, value) =>
+      dataKey === TREND_LINE_DATA_KEY ? applySquareRootScaling(value) : value,
+    ),
+  ) as TrendDataset[];
+}
+
+/**
+ * Computes the dataset and series option objects needed by ECharts to render trend lines.
+ * Each series in the chart model will have one corresponding dataset and option, in the same
+ * order as the `chartModel.seriesModels` array.
+ *
+ * @param {CartesianChartModel} chartModel - Local computed data model for the chart
+ * @param {ComputedVisualizationSettings} settings - Locally computed visualization settings for the chart
+ * @param {RenderingContext} renderingContext - Misc. helpers provided by the environment (client or static-viz pipeline)
+ * @returns Object ({ options, datasets }), each key is either `null` or an array
+ */
+export function getTrendLineOptionsAndDatasets(
+  chartModel: CartesianChartModel,
+  settings: ComputedVisualizationSettings,
+  renderingContext: RenderingContext,
+): {
+  options: RegisteredSeriesOption["line"][] | null;
+  datasets: EChartsOption["dataset"][] | null;
+} {
+  // The trend line can only be shown when the question has only
+  // a single aggregation on a time field. When this is not the case,
+  // the backend will return null for the insights object, so our array
+  // will have length 0.
+  const canShowTrendLine = chartModel.insights.length !== 0;
+  if (!settings["graph.show_trendline"] || !canShowTrendLine) {
+    return { options: null, datasets: null };
+  }
+  if (chartModel.insights.length !== chartModel.seriesModels.length) {
+    throw Error("Number of insight objects does not match number of series");
+  }
+
+  // series option objects for each trend line (one per series)
+  const options: RegisteredSeriesOption["line"][] = chartModel.seriesModels.map(
+    (seriesModel, index) => ({
+      type: "line",
+      datasetIndex: index + 1, // offset to account for the chart's dataset (e.g. question results)
+      encode: {
+        x: chartModel.dimensionModel.dataKey,
+        y: TREND_LINE_DATA_KEY,
+      },
+      showSymbol: false,
+      lineStyle: {
+        color: Color(renderingContext.getColor(seriesModel.color))
```

Comment:
Using `Color` directly instead of the existing `lighten` method because it returns an `hsl` string, which Batik does not support